azurerm/internal/services/storage/resource_arm_storage_blob.go

```diff
@@ -233,9 +251,21 @@ func resourceArmStorageBlobUpdate(d *schema.ResourceData, meta interface{}) erro
 
 	if d.HasChange("content_type") {
 		log.Printf("[DEBUG] Updating Properties for Blob %q (Container %q / Account %q)...", id.BlobName, id.ContainerName, id.AccountName)
+		// `content_md5` is `ForceNew` but must be included in the `SetPropertiesInput` update payload or it will be zeroed on the blob.
+		contentMD5 := d.Get("content_md5").(string)
+		if contentMD5 != "" {
+			data, err := convertHexToBase64Encoding(contentMD5)
+			if err != nil {
+				return fmt.Errorf("Error in converting hex to base64 encoding for content_md5: %s", err)
+			}
+			contentMD5 = data
+		}
+
 		input := blobs.SetPropertiesInput{
 			ContentType: utils.String(d.Get("content_type").(string)),
+			ContentMD5:  utils.String(contentMD5),
```

Comment:
(same here) for the sake of completeness, does this want to be:

```suggestion
		}
		
		if contentMD5 != "" {
			input.ContentMD5:  utils.String(contentMD5),
		}
```